frontend/src/metabase/admin/permissions/utils/graph/data-permissions.unit.spec.ts

```diff
@@ -0,0 +1,216 @@
+        ),
+      ).toBe(false);
+    });
+
+    it("should handle databases with multiple schemas", async () => {
+      const schemas = [
+        { name: "public", getTables: () => [{ id: 1 }] },
+        { name: "public2", getTables: () => [{ id: 2 }] },
+      ];
+      const database = {
+        schemas,
+        schema: (name: string) => schemas.find(schema => schema.name === name),
+      };
+
+      const testPermissions: GroupsPermissions = {
+        "1": {
+          "1": {
+            [DataPermission.VIEW_DATA]: {
+              public: DataPermissionValue.UNRESTRICTED,
+              public2: DataPermissionValue.LEGACY_NO_SELF_SERVICE,
+            },
+          },
+        },
+      };
+
+      const testFn = _.partial(
+        hasPermissionValueInSubgraph,
+        testPermissions,
+        1,
+        { databaseId: 1 },
+        database,
+        DataPermission.VIEW_DATA,
+      );
+
+      expect(testFn(DataPermissionValue.UNRESTRICTED)).toBe(true);
+      expect(testFn(DataPermissionValue.LEGACY_NO_SELF_SERVICE)).toBe(true);
+      expect(testFn(DataPermissionValue.BLOCKED)).toBe(false);
+    });
+
+    it("should handle schema entity ids", async () => {
+      const schemas = [
+        { name: "public", getTables: () => [{ id: 1 }] },
+        { name: "public2", getTables: () => [{ id: 2 }] },
+      ];
+      const database = {
+        schemas,
+        schema: (name: string) => schemas.find(schema => schema.name === name),
+      };
+
+      const testPermissions: GroupsPermissions = {
+        "1": {
+          "1": {
+            [DataPermission.VIEW_DATA]: {
+              public: DataPermissionValue.UNRESTRICTED,
+              public2: DataPermissionValue.BLOCKED,
+            },
+          },
+        },
+      };
+
+      expect(
+        hasPermissionValueInSubgraph(
+          testPermissions,
+          1,
+          { databaseId: 1, schemaName: "public" },
+          database,
+          DataPermission.VIEW_DATA,
+          DataPermissionValue.UNRESTRICTED,
+        ),
+      ).toBe(true);
+
+      expect(
+        hasPermissionValueInSubgraph(
+          testPermissions,
+          1,
+          { databaseId: 1, schemaName: "public2" },
+          database,
+          DataPermission.VIEW_DATA,
+          DataPermissionValue.BLOCKED,
+        ),
+      ).toBe(true);
+
+      expect(
+        hasPermissionValueInSubgraph(
+          testPermissions,
+          1,
+          { databaseId: 1, schemaName: "public" },
+          database,
+          DataPermission.VIEW_DATA,
+          DataPermissionValue.BLOCKED,
+        ),
+      ).toBe(false);
+
+      expect(
+        hasPermissionValueInSubgraph(
+          testPermissions,
+          1,
+          { databaseId: 1, schemaName: "public2" },
+          database,
+          DataPermission.VIEW_DATA,
+          DataPermissionValue.UNRESTRICTED,
+        ),
+      ).toBe(false);
+    });
+
+    it("should permissions omited from the graph", async () => {
```

Comment:
```suggestion
    it("should permissions be omitted from the graph", async () => {
```

Reply:
ooops, this should have been `"should handle default permissions omitted from the graph"`